src/librustc/infer/error_reporting.rs

```diff
@@ -554,7 +554,7 @@ impl<'a, 'tcx> ErrorReporting<'tcx> for InferCtxt<'a, 'tcx> {
                                        "{}",
```

Comment:
so, what I was trying to say before was that if we wanted to have something more like the old output, we could remove the modification below and instead change this line to:

```
let message = if check_old_skool() {
    format("{} ({})", trace.origin, terr)
} else {
    format("{}", trace.origin)
};
let mut err = struct_span_err!(..., "{}", message);
```

But I guess I'm ok either way.